documentacion/phpDoc/man/man3/Invitado.3
.TH "Invitado" 3 "Viernes, 20 de Mayo de 2022" "Version V18" "Parque Ecológico" \" -*- nroff -*-
.ad l
.nh
.SH NAME
Invitado
.SH SYNOPSIS
.br
.PP
.PP
Herencias class y \fBPersona\fP\&.
.SS "Métodos públicos"

.in +1c
.ti -1c
.RI "\fB__construct\fP ($usuario, $nombreApellido, $mail, $claveAcceso)"
.br
.ti -1c
.RI "\fBgetClaveAcceso\fP ()"
.br
.ti -1c
.RI "\fBsetClaveAcceso\fP ($claveAcceso)"
.br
.ti -1c
.RI "\fBgetUsuario\fP ()"
.br
.ti -1c
.RI "\fBsetUsuario\fP ($usuario)"
.br
.ti -1c
.RI "\fBgetNombreApellido\fP ()"
.br
.ti -1c
.RI "\fBsetMail\fP ($mail)"
.br
.ti -1c
.RI "\fBsetNombreApellido\fP ($nombreApellido)"
.br
.ti -1c
.RI "\fBmostrarInformacion\fP ()"
.br
.ti -1c
.RI "\fBeditarInformacion\fP ()"
.br
.in -1c
.SH "Descripción detallada"
.PP 
Proyecto Parque Eocológico
.PP
\fBVersión\fP
.RS 4
pq-_V18 @autor Virginia M\&. García Álvarez Clase \fBInvitado\fP
.RE
.PP
persona 
.PP
Definición en la línea \fB16\fP del archivo \fBInvitado\&.php\fP\&.
.SH "Documentación del constructor y destructor"
.PP 
.SS "__construct ( $usuario,  $nombreApellido,  $mail,  $claveAcceso)"
Constructor de 4 parámetros; crea y devuelve el objeto \fBInvitado\fP a partir de los datos solicitados
.PP
\fBParámetros\fP
.RS 4
\fI$usuario\fP 
.br
\fI$nombreApellido\fP 
.br
\fI$mail\fP 
.br
\fI$claveAcceso\fP 
.RE
.PP
\fBDevuelve\fP
.RS 4
\fBInvitado\fP 
.RE
.PP

.PP
Reimplementado de \fBPersona\fP\&.
.PP
Definición en la línea \fB35\fP del archivo \fBInvitado\&.php\fP\&.
.SH "Documentación de las funciones miembro"
.PP 
.SS "editarInformacion ()"
Permite la recogida de datos para la modificación de los mismos\&. 
.PP
Reimplementado de \fBPersona\fP\&.
.PP
Definición en la línea \fB116\fP del archivo \fBInvitado\&.php\fP\&.
.SS "getClaveAcceso ()"

.PP
\fBDevuelve\fP
.RS 4
string la clave de acceso introducida 
.RE
.PP

.PP
Definición en la línea \fB43\fP del archivo \fBInvitado\&.php\fP\&.
.SS "getNombreApellido ()"

.PP
\fBDevuelve\fP
.RS 4
string el nombre introducido 
.RE
.PP

.PP
Reimplementado de \fBPersona\fP\&.
.PP
Definición en la línea \fB71\fP del archivo \fBInvitado\&.php\fP\&.
.SS "getUsuario ()"

.PP
\fBDevuelve\fP
.RS 4
string el nombre de usuario introducido 
.RE
.PP

.PP
Definición en la línea \fB57\fP del archivo \fBInvitado\&.php\fP\&.
.SS "mostrarInformacion ()"
Muestra la información del objeto 
.PP
Reimplementado de \fBPersona\fP\&.
.PP
Definición en la línea \fB93\fP del archivo \fBInvitado\&.php\fP\&.
.SS "setClaveAcceso ( $claveAcceso)"
Modifica la clave introducida
.PP
\fBParámetros\fP
.RS 4
\fI$claveAcceso\fP 
.RE
.PP

.PP
Definición en la línea \fB51\fP del archivo \fBInvitado\&.php\fP\&.
.SS "setMail ( $mail)"
Modifica el mail introducido
.PP
\fBParámetros\fP
.RS 4
\fI$mail\fP 
.RE
.PP

.PP
Reimplementado de \fBPersona\fP\&.
.PP
Definición en la línea \fB79\fP del archivo \fBInvitado\&.php\fP\&.
.SS "setNombreApellido ( $nombreApellido)"
Modifica el nombre y apellido introducido
.PP
\fBParámetros\fP
.RS 4
\fI$nombreApellido\fP 
.RE
.PP

.PP
Reimplementado de \fBPersona\fP\&.
.PP
Definición en la línea \fB87\fP del archivo \fBInvitado\&.php\fP\&.
.SS "setUsuario ( $usuario)"
Modifica el nombre de usuario
.PP
\fBParámetros\fP
.RS 4
\fI$usuario\fP 
.RE
.PP

.PP
Definición en la línea \fB65\fP del archivo \fBInvitado\&.php\fP\&.

.SH "Autor"
.PP 
Generado automáticamente por Doxygen para Parque Ecológico del código fuente\&.


php/clasesPHP/persona/Invitado.php
<?php

/**
 * Proyecto Parque Eocológico
 * 
 * @version pq-_V18
 * @autor Virginia M. García Álvarez
 */


/**
 * Clase Invitado
 * 
 * @extends class persona
 */
class Invitado extends Persona{
    /**
     * @var string Nombre de usuario
     */
    private $usuario;
    /**
     * @var string clave de acceso al sitio web
     */
    private $claveAcceso;
    /**
     * Constructor de 4 parámetros; crea y devuelve el objeto Invitado a partir de los datos solicitados
     * 
     * @param strig $usuario
     * @param string $nombreApellido
     * @param string $mail
     * @param string $claveAcceso
     * 
     * @return Invitado
     */
    function __construct($usuario, $nombreApellido, $mail, $claveAcceso) {
        parent::__construct($nombreApellido, null, $mail, null);
        $this->usuario = $usuario;
        $this->claveAcceso = $claveAcceso;
    }
    /**
     * @return string la clave de acceso introducida
     */
    function getClaveAcceso() {
        return $this->claveAcceso;
    }
    /**
     * Modifica la clave introducida
     * 
     * @param string $claveAcceso
     */
    function setClaveAcceso($claveAcceso) {
        $this->claveAcceso = $claveAcceso;
    }
    /**
     * @return string el nombre de usuario introducido
     */
    function getUsuario() {
        return $this->usuario;
    }
    /**
     * Modifica el nombre de usuario
     * 
     * @param string $usuario
     */
    function setUsuario($usuario) {
        $this->usuario = $usuario;
    }
    /**
     * @return string el nombre introducido
     */
    public function getNombreApellido() {
        return parent::getNombreApellido();
    }
    /**
     * Modifica el mail introducido
     * 
     * @param string $mail
     */
    public function setMail($mail) {
        parent::setMail($mail);
    }
    /**
     * Modifica el nombre y apellido introducido
     * 
     * @param string $nombreApellido
     */
    public function setNombreApellido($nombreApellido) {
        parent::setNombreApellido($nombreApellido);
    }
    /**
     * Muestra la información del objeto
     */
    public function mostrarInformacion() {
    ?>
        <table id="tablaI">
            <tbody id="tbodyI">
                <tr>
                    <td>Nombre: </td>
                    <td><?php echo parent::getNombreApellido();?></td>
                </tr>
                <tr>
                    <td>Nombre de usuario: </td>
                    <td><?php echo $this->usuario;?></td>
                </tr>
                <tr>
                    <td>Email: </td>
                    <td><?php echo parent::getMail();?></td>
                </tr>
            </tbody>
        </table>
    <?php
    }
    /**
     * Permite la recogida de datos para la modificación de los mismos.
     */
    public function editarInformacion() {
    ?>
        <p><label for="nombre"/>Nombre: 
        <input type="text" id="nombre" placeholder="<?php echo parent::getNombreApellido();?>" autocomplete="off"/></p>
        <p><label for="usuario"/>Usuario: <?php echo $this->usuario;?>
        <p><label for="mail"/>Mail: 
        <input type="text" id="mail" placeholder="<?php echo parent::getMail();?>" autocomplete="off"/></p>
        <p><label for="claveM"/>Clave: 
        <input type="password" id="claveM" placeholder="contraseña"  /></p>
    <?php
    }

}
